File: artifacts/api-server/python/routers/tags.py
from fastapi import APIRouter
from db import db

router = APIRouter()

@router.get("/tags")
def list_tags():
    with db() as conn:
        cur = conn.cursor()
        cur.execute("SELECT * FROM tags ORDER BY name")
        tags = cur.fetchall()
        result = []
        for tag in tags:
            cur.execute(
                "SELECT COUNT(*) as cnt FROM request_tags WHERE tag_id = %s", (tag["id"],)
            )
            count = cur.fetchone()["cnt"]
            result.append({
                "id": tag["id"],
                "name": tag["name"],
                "color": tag["color"],
                "requestCount": count,
            })
        return result

File: artifacts/api-server/src/routes/tags.ts
import { Router, type IRouter } from "express";
import { count, eq } from "drizzle-orm";
import { db, tagsTable, requestTagsTable } from "@workspace/db";

const router: IRouter = Router();

router.get("/tags", async (_req, res): Promise<void> => {
  const tags = await db.select().from(tagsTable).orderBy(tagsTable.name);

  const tagsWithCounts = await Promise.all(tags.map(async (tag) => {
    const [result] = await db.select({ count: count() }).from(requestTagsTable).where(eq(requestTagsTable.tagId, tag.id));
    return {
      id: tag.id,
      name: tag.name,
      color: tag.color,
      requestCount: Number(result?.count ?? 0),
    };
  }));

  res.json(tagsWithCounts);
});

export default router;
